### PR TITLE
added com04-aws200 to new idea

### PR DESCRIPTION
**Pull request by KaranGauswami**
Added new idea as per #19

### CODE REVIEW

**Review by madebygps — CHANGES_REQUESTED** [review]
COM04-AWS200.md looks good, can you also add the entry to the correct table in the readme.md

**Comment by KaranGauswami** [comment]
@madebygps Applied changes.

File: Projects/COM/COM04/COM04-AWS200.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- [Get Started with Docker Swarm](https://docs.docker.com/engine/swarm/swarm-tutorial)
+
+## Costs
+
+- Included in the Free Tier
+- \$1-2 per hour running
+
+## Estimated time to complete
+
+- 30-45 minutes
```
Comment (what-name) [inline Projects/COM/COM04/COM04-AWS200.md]: Time to complete is underestimated. Please wait with merging this PR, I will do this project and report back on how long it takes approx. I think this is in the ballpark of 3-4 hours, especially for someone not too familiar with Docker or Swarms. I think we have to be honest about time required.

_"The project said it takes 30 minutes, I've been at this for 5 hours...what is happening"_ - if you know what I mean.

**Review by what-name — COMMENTED** [review]
Wait with merge, time required is underestimated.

File: Projects/COM/COM04/COM04-AWS200.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Project's Author(s)
+
+[Karan Gauswami](https://github.com/KaranGauswami)
+
+## Objectives
+
+### You need to complete the following:
+
+- Create a VPC for our EC2 Instances
```
Comment (what-name) [inline Projects/COM/COM04/COM04-AWS200.md]: Why is a new VPC needed? There is a VPC in place by default on AWS. There is no harm in using the default VPC for this. Projects should overlap only when really needed (COM-NET). New VPC means new route tables, new IGW and new Subnets all have to be created manually.

File: Projects/COM/COM04/COM04-AWS200.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- Create an EC2 Instance with initial script to install Docker
+- Setup that EC2 Instance as a manager node
+- Create multiple instances (4 to 5) with initial script to create worker nodes and join that manager node
```
Comment (what-name) [inline Projects/COM/COM04/COM04-AWS200.md]: - 2-3 instances should be enough. As long as above two, the number of instances does not influence the amount of learning achieved. The big number required can scare away people $$$ (this was my first thought too when I read it).
- "Initial script" is called "User Data" on AWS, very important detail - If that's what it refers to.

~~Everything else looks good👍~~

File: Projects/COM/COM04/COM04-AWS200.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# COM04-AWS200 - Create a cluster of virtual machines using docker swarm
+
+## Cloud Service Provider
+
+- Amazon Web Services
+
+## Difficulty
+
+- Level 200 (Intermediate)
```
Comment (antoniolofiego) [inline Projects/COM/COM04/COM04-AWS200.md]: Anything Docker Swarm related is not 200 level. This is easily 300 if not 400, it requires extensive experience not only in Docker itself but the whole swarm concept. This is material at the level of the AWS Certified DevOps Professional certificate.

For a reference on a 200 level, check https://github.com/100DaysOfCloud/100DaysOfCloudIdeas/blob/master/Projects/SEC/SEC01/SEC01-AWS200.md.

File: Projects/COM/COM04/COM04-AWS200.md (hunk 1 above)
Comment (antoniolofiego) [reply Projects/COM/COM04/COM04-AWS200.md]: I agree with this. This timing is severely underestimated and can be done in 30-45 minutes only by an experience DevOps/Sys Admin.

File: Projects/COM/COM04/COM04-AWS200.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+## Estimated time to complete
+
+- 30-45 minutes
+
+## Tips
+
+- Use t2.micro instances for creating nodes to avoid unexpected charges.
+- To add more Manager or Worker node , It is not required to SSH into that Instance.
+- Open SSH port for only required machines in Securiy groups.
```
Comment (antoniolofiego) [inline Projects/COM/COM04/COM04-AWS200.md]: Fix "Security Group" typo.

**Review by antoniolofiego — CHANGES_REQUESTED** [review]

File: Projects/COM/COM04/COM04-AWS200.md (hunk 5 above)
Comment (what-name) [reply Projects/COM/COM04/COM04-AWS200.md]: - To add more manager or worker nodes, it is not required to SSH into the new instance. Utilize the "user data" functionality.
- Only allow incoming SSH traffic in security groups for the instances that require it.

Phrasing + typos.

File: Projects/COM/COM04/COM04-AWS200.md (hunk 4 above)
Comment (what-name) [reply Projects/COM/COM04/COM04-AWS200.md]: Agreed. This project involves writing complicated user data scripts that set up EC2 nodes, Docker[Swarm] knowledge and NGinx or similar web server. Also knowledge on how to prove that the swarm is indeed working. Not sure if 300 or 400 but def. not 200.

**Review by what-name — CHANGES_REQUESTED** [review]
To summarize:
- Difficulty level should be either 300 or 400. I'm leaning to 400 because of the user data part.
- Typos and phrasing issues where mentioned before.
- Omit the need for a new VPC, not the point of this project.
- Time required is underestimated.
- Worker node instances can be 2-3. Current 4-5 is unnecessary.

**Review by madebygps — CHANGES_REQUESTED** [review]
level 300 at least and minimum of 1 hour work here.

File: Projects/COM/COM04/COM04-AWS200.md (hunk 1 above)
Comment (KaranGauswami) [reply Projects/COM/COM04/COM04-AWS200.md]: Added estimated  time to 4-5 hours.

File: Projects/COM/COM04/COM04-AWS200.md (hunk 2 above)
Comment (KaranGauswami) [reply Projects/COM/COM04/COM04-AWS200.md]: Removed VPC

File: Projects/COM/COM04/COM04-AWS200.md (hunk 3 above)
Comment (KaranGauswami) [reply Projects/COM/COM04/COM04-AWS200.md]: - Changed Instance count from 4-5 to 2-3
- Replaced initial script with User data script.

File: Projects/COM/COM04/COM04-AWS200.md (hunk 5 above)
Comment (KaranGauswami) [reply Projects/COM/COM04/COM04-AWS200.md]: Fixed Phrasing and typos

File: Projects/COM/COM04/COM04-AWS400.md [hunk 6]
```diff
@@ -16,18 +16,17 @@
...
+- Create multiple instances (2 to 3) with user data script to create worker nodes and join that manager node
 - Start any service like nginx to test our Swarm Cluster
 - Scale that service to multiple nodes
 - Delete all the resources you created after completion
 
 ### You need to answer the following:
 
 - How to check status of Docker nodes ?
-- What behaviour is expected if one of the Worker or Manager node went down ?
+- What behaviour is expected if one of the worker or manager node went down ?
```
Comment (what-name) [inline Projects/COM/COM04/COM04-AWS400.md]: - "nodes" is plural.

**Review by what-name — CHANGES_REQUESTED** [review]
- One change, see comment.

Everything else looks great👍 After that added "s", PR approved from my end.
* * *
**Heads up:** Lets squash the commits.

**Review by madebygps — CHANGES_REQUESTED** [review]
Node should be plural per @what-name , after that PR is approved on my end.

**Comment by KaranGauswami** [comment]
Applied changes.

File: Projects/COM/COM04/COM04-AWS400.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+- Create multiple instances (2 to 3) with user data script to create worker nodes and join them with manager node
+- Start any service like nginx to test our Swarm Cluster
+- Scale that service to multiple nodes
+- Delete all the resources you created after completion
+
+### You need to answer the following:
+
+- How to check status of Docker nodes ?
+- What behaviour is expected if one of the worker or manager nodes went down ?
+- How to get state of all the containers running on diffrent nodes ?
```
Comment (what-name) [inline Projects/COM/COM04/COM04-AWS400.md]: "diffrent" -> "different"

Comment (KaranGauswami) [reply Projects/COM/COM04/COM04-AWS400.md]: Fixed.

**Review by madebygps — APPROVED** [review]
Looks good, thank you :)

**Pull request merged**